watch-parties/src/main/java/com/google/sps/servlets/ChatListerServlet.java

```diff
@@ -13,21 +27,25 @@
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
+//import atg.repository.QueryOptions;
 import com.google.appengine.api.datastore.Query.SortDirection;
 import java.util.Date;
 import com.google.sps.data.Chat;
+import java.util.LinkedList; 
+import java.util.Stack; 
 
 
 @WebServlet("/chatstorage")
 public class ChatListerServlet extends HttpServlet {
 
   public ArrayList<Chat> comments = new ArrayList<>();
+  public Stack<Chat> stackComments = new Stack<>();
```

Comment:
In fact, since we're reading from the Datastore for every chat request, you can just build the response as you iterate through the query results. Github won't let me make a code suggestion for this method, so I copied it into a gist here: https://gist.github.com/Sunlis/6088ca5fa58a004dbb310673d92fadc5

Reply:
This should also remove the need for the `written` checks you're doing.